src/parser/ASTBuilder.h

```diff
@@ -142,34 +142,44 @@ class SyntaxNode {
     static void* operator new(size_t, void* p) { return p; } // for VectorWithInlineStorage
     SyntaxNode()
         : m_nodeType(ASTNodeTypeError)
+        , m_flag(false)
```

Comment:
It seems that `m_flags` is used only for `ArrowParameterPlaceHolder` node type. How about merging `m_flag` and `m_string` members into union variable?

Reply:
Yes, you are right.
I consider 2 idea(using union or making new member like PR)
I choose second idea because it can be used by other place & easy to understand.

Reply:
I see. We can consider using union approach later if we have to use memory more tightly.